### ISSUE TITLE
🐞 dagger.io/dagger v0.11.7 is not available through dagger.io

### ISSUE THREAD

**Issue by kjuulh** [dagger/dagger#7630]
### What is the issue?

I think your proxy was left out in the cold when you pushed the latest release to github

### Dagger version

dagger v0.11.7

### Steps to reproduce

go get dagger.io/dagger@v0.11.7 => dagger.io/dagger v0.11.6

### Log output

```
❯ go get dagger.io/dagger@v0.11.7
go: downloading dagger.io/dagger v0.11.7
go: dagger.io/dagger@v0.11.7 requires dagger.io/dagger@v0.11.7: parsing go.mod:
        module declares its path as: github.com/dagger/dagger
                but was required as: dagger.io/dagger
```

_No response_

**Comment by morlay** [comment]
https://proxy.golang.org/dagger.io/dagger/@v/v0.11.7.mod
https://proxy.golang.org/github.com/dagger/dagger/@v/v0.11.7.mod

seems `github.com/dagger/dagger` is released as `dagger.io/dagger`

**Comment by jedevc** [comment]
Investigating now.

Looks like https://github.com/dagger/dagger-go-sdk, the source of truth for this is wrong.

**Comment by kjuulh** [comment]
@jedevc it still seems broken:

```
❯ go get dagger.io/dagger@v0.11.7
go: dagger.io/dagger@v0.11.7 requires dagger.io/dagger@v0.11.7: parsing go.mod:
        module declares its path as: github.com/dagger/dagger
                but was required as: dagger.io/dagger
```

**Comment by jedevc** [comment]
Yeah, sorry we still need to repush the v0.11.7 tag - @gerhard is looking into it.

**Comment by sagikazarmark** [comment]
Maybe tag a new version instead? Go is quite sensitive to changing tags.

**Comment by gerhard** [comment]
The Go SDK has now been re-tagged: https://github.com/dagger/dagger-go-sdk/releases/tag/v0.11.7

This is the expected version & directory structure: https://github.com/dagger/dagger-go-sdk/tree/443305dc487d22254940e3029bd42d1b45de2dcc

<img width="1552" alt="image" src="https://github.com/dagger/dagger/assets/3342/1d631da7-abc0-4ff4-a323-be8d3841772c">

And this is the expected `go.sum`: https://github.com/dagger/dagger-go-sdk/blob/443305dc487d22254940e3029bd42d1b45de2dcc/go.mod

With all that in place, I am still getting the following error - as captured in this issue:

```console
❯ go get dagger.io/dagger@v0.11.7
go: dagger.io/dagger@v0.11.7 requires dagger.io/dagger@v0.11.7: parsing go.mod:
	module declares its path as: github.com/dagger/dagger
	        but was required as: dagger.io/dagger
```

I think this is happening because https://pkg.go.dev/dagger.io/dagger@v0.11.7 is not in the right shape. Still investigating.

**Comment by aluzzardi** [comment]
`dagger/dagger-go-sdk` is a "mirror" of `github.com/dagger/dagger/sdk/go`.

The magic happens here: https://github.com/dagger/dagger/blob/main/ci/sdk_go.go#L67-L100

Can't check right away but maybe there have been recent modifications?

**Comment by matipan** [comment]
If we start a module from scratch, clean local mod cache (to reduce potential noise) and get `dagger.io/dagger@v0.11.7` without going through the proxy (`GOPROXY=direct`), we can see that the issue is due to a checksum missmatch. Given how the proxy works, it seems that the v0.11.7 will remain broken until it is cleared from sum.golang.org (which I'm not sure if its possible to do). You should be able to repro this with the following steps:

```terminal
$ mkdir dagger-v0-11-7 && cd dagger-v0-11-7
$ go mod init testing
$ go clean -modcache
$ export GOPROXY=direct
$ go get -u dagger.io/dagger@v0.11.7
go: downloading dagger.io/dagger v0.11.7
go: dagger.io/dagger@v0.11.7: verifying module: checksum mismatch
        downloaded: h1:5KwB14CNJ1WdW5MEKZwsIrepaQcXKmCAQEQNJgy5f5c=
        sum.golang.org: h1:xXa7dgyM5xGZlWC/uc9mlRIH2XrTZenemTM/2vfibb4=

SECURITY ERROR
This download does NOT match the one reported by the checksum server.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

![image](https://github.com/dagger/dagger/assets/8126891/e5fdbc53-1c9b-485f-b494-483cd8adf5a6)

**Comment by gerhard** [comment]
We had to update (force push) the `v0.11.7` tag due to:
- https://github.com/dagger/dagger/pull/7631

Considering that we discovered a few edge cases in our CI that we want to dig into, it sounds reasonable to follow-up with `v0.11.8` and a few of these fixes. FTR: https://discord.com/channels/707636530424053791/1248289310105669734/1250122548633669663

**Comment by gerhard** [comment]
`go get dagger.io/dagger@v0.11.8-rc.2` is now available & it will work exactly as `v0.11.7` would have.

We have hit a different problem with `v0.11.8-rc.1` as captured here:
- https://github.com/dagger/dagger/pull/7642#issuecomment-2163684518

We will need to release another RC - most probably `rc.3` - so that the release notes line up & everything looks right, but we ran out of time today.

**Comment by gerhard** [comment]
Thank you for reporting this @kjuulh 💪

https://proxy.golang.org/dagger.io/dagger/@v/v0.11.8-rc.2.mod

![image](https://github.com/dagger/dagger/assets/3342/1753a94c-2fc8-409f-80a4-ebf9e6c78d42)
